datalog/src/main/java/org/wpilib/datalog/DataLogRecord.java

```diff
@@ -143,7 +143,7 @@ public static class StartRecordData {
     public final String type;
 
     /** Initial metadata. */
-    public final String metadata;
+    public String metadata;
```

Comment:
It feels really weird to me to make this mutable... Maybe instead of modifying `readerEntry.metadata`, we update `m_entriesById` and `m_entriesByName` with a new `DataLogReaderEntry`?